File: samples/CS/15-ContextPopup/RibbonContextMenuStrip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.ComponentModel;
using WinForms.Ribbon;

namespace WinForms.Ribbon
{
    public class RibbonContextMenuStrip : ContextMenuStrip
    {
        private uint _contextPopupID;
        private RibbonStrip _ribbon;

        public RibbonContextMenuStrip(RibbonStrip ribbon, uint contextPopupID)
            : base()
        {
            _contextPopupID = contextPopupID;
            _ribbon = ribbon;
        }

        protected override void OnOpening(CancelEventArgs e)
        {
            _ribbon.ShowContextPopup(_contextPopupID, Cursor.Position.X, Cursor.Position.Y);
            e.Cancel = true;
        }
    }
}

File: samples/VB/15-ContextPopup/RibbonContextMenuStrip.vb
Imports System.Text
Imports System.ComponentModel
Imports WinForms.Ribbon

Namespace _15_ContextPopup
	Public Class RibbonContextMenuStrip
		Inherits ContextMenuStrip
		Private _contextPopupID As UInteger
		Private _ribbon As RibbonStrip

		Public Sub New(ByVal ribbon_Renamed As RibbonStrip, ByVal contextPopupID As UInteger)
			MyBase.New()
			_contextPopupID = contextPopupID
			_ribbon = ribbon_Renamed
		End Sub

		Protected Overrides Sub OnOpening(ByVal e As CancelEventArgs)
			_ribbon.ShowContextPopup(_contextPopupID, Cursor.Position.X, Cursor.Position.Y)
			e.Cancel = True
		End Sub
	End Class
End Namespace
